main/read.c

```diff
@@ -135,6 +135,13 @@ extern unsigned long getInputLineNumber (void)
 	return File.input.lineNumber;
 }
 
+extern int getInputLineOffset (void)
+{
+	unsigned char *base = (unsigned char *) vStringValue (File.line);
+	int ret = File.currentLine - base - File.ungetchIdx;
+	return ret >= 0 ? ret : 0;
```

Comment:
Right, I noticed that too but it shouldn't be a practical problem as one typically ungec()s either real code characters (new token) or whitespace characters (EOLs to stay at the previous line) but not both of them at the same time. So there shouldn't be any forgotten non-whitespace characters from the previous line and we'll only miss some whitespace characters which don't matter.